lib/improver/tests/utilities/test_cube_extraction.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+""" Unit tests for cube extraction utilities """
+
+import unittest
+import os
+import numpy as np
+from tempfile import mkdtemp
+from subprocess import call
+
+import iris
+from iris.tests import IrisTest
+from iris.coords import DimCoord
+from iris.exceptions import CoordinateNotFoundError
+
+from improver.utilities.cube_extraction import (parse_constraint_list,
+                                                extract_subcube)
+
+
+def set_up_precip_probability_cube():
+    """
+    Set up a cube with spatial probabilities of precipitation at three
+    exceedance thresholds
+    """
+    data = np.array([[[0.85, 0.95, 0.73],
+                      [0.75, 0.85, 0.65],
+                      [0.70, 0.80, 0.62]],
+                     [[0.18, 0.20, 0.15],
+                      [0.11, 0.16, 0.09],
+                      [0.10, 0.14, 0.03]],
+                     [[0.03, 0.04, 0.01],
+                      [0.02, 0.02, 0.00],
+                      [0.01, 0.00, 0.00]]])
+
+    mmh_to_ms = 0.001/3600.
```

Comment:
(Unimportant given this is only a unit test, but it's a good idea to have constants in caps, so we know they don't change.)